### PR TITLE
add information from drug_strength table

### PR DESCRIPTION
**Pull request by Xintong-Li-ZnCu**
when domains = "drug"

### CODE REVIEW

File: R/runSearch.R [hunk 1]
```diff
@@ -137,6 +138,20 @@ runSearch <- function(keywords,
       dplyr::rename_with(tolower)
   }
 
+  # collect the drug_strength table if drug
+  if (domains == "drug"){
```
Comment (edward-burn) [inline R/runSearch.R:142]: should this be %in% ? The user can search for codes in multiple domains at the same time. What do we want the behaviour to be if the user searches both conditions and drugs at the same time?

File: R/runSearch.R [hunk 2]
```diff
@@ -475,18 +490,41 @@ runSearch <- function(keywords,
       ) %>%
       dplyr::distinct()
 
+    candidateCodesOrig <- candidateCodes  #  temp save the list
+
+    #if domain = "drug", add drug_strength information
+
+    if (domains=="drug"){
```
Comment (edward-burn) [inline R/runSearch.R:497]: As above, %in% ?

File: R/runSearch.R [hunk 3]
```diff
@@ -475,18 +490,41 @@ runSearch <- function(keywords,
...
         "vocabulary_id", "found_from"
       )
+    }
 
     candidateCodes <- candidateCodes %>%
       dplyr::distinct()
 
     # remove duplicates (found in different ways)
     # keep first time it was found
+    # for drug,  same concept_id with different ingredient_concept_id will be removed as well.
```
Comment (edward-burn) [inline R/runSearch.R:527]: not sure I follow - what happens here now?

**Comment by edward-burn** [comment]
thanks for this @Xintong-Li-ZnCu. As well as the above comments, can you also add some tests of this please? For example, checking the behaviour if the user searches for multiple domains etc. 

To add tests with the mock you would I suppose have to also add drug strength in the mock https://github.com/darwin-eu/CodelistGenerator/blob/5ea108d3a0357d4f03483d508c55131264480ffa/R/mockVocabRef.R#L31. But maybe easier would be to add your tests with Eunomia at least to start with https://github.com/darwin-eu/CodelistGenerator/blob/5ea108d3a0357d4f03483d508c55131264480ffa/tests/testthat/test-eunomia.R#L3

File: R/runSearch.R (hunk 1 above)
Comment (Xintong-Li-ZnCu) [reply R/runSearch.R:142]: https://github.com/darwin-eu/CodelistGenerator/blob/3598abc543c83647963ae4ac304141c28ba22d27/R/getCandidateCodes.R#L171-L179

But with these, to my understand, the `domains` goes into the `runSearch` function will always be one?  unless the user run the `runSearch` function directly?   But I can change it to `"drug" %in% tolower(domains) `

Comment (edward-burn) [reply R/runSearch.R:142]: Ah yes, you´re right!

File: R/runSearch.R (hunk 3 above)
Comment (Xintong-Li-ZnCu) [reply R/runSearch.R:527]: in the drug_strength table, one drug_concept_id can have multiple lines with different ingredient_concept_id, so if we don't necessary need this information this doesn't matter .

**Comment by edward-burn** [comment]
thanks @Xintong-Li-ZnCu can you see if my minor tweaks look ok (https://github.com/darwin-eu/CodelistGenerator/pull/43/commits/0e493fb7dfa51872652d29cb7b3ff3c879ad030a)? If so I think this is good to merge

**Comment by Xintong-Li-ZnCu** [comment]
>

**Comment by Xintong-Li-ZnCu** [comment]
> 

yes I think it's good to go

**Pull request merged**